### PR TITLE
Use bazel to create boskos images

### PR DESCRIPTION
**Pull request by sebastienvas**
I think this was a long overdue. I am currently working on Boskos and decided to start with that to ease development.

### CODE REVIEW

**Comment by sebastienvas** [comment]
@Katharine do I need to change the cloudbuild.yaml ?

**Comment by krzyzacy** [comment]
glad to see this happen! punt to image experts:

/assign @Katharine @fejta

**Comment by Katharine** [comment]
> @Katharine do I need to change the cloudbuild.yaml ?

You should probably just remove them, as well as `boskos/variants.yaml` (though before doing so, also consider what to do with `aws-janitor`, which you have not touched here).

You'll also want to remove the [`post-test-infra-push-boskos` job](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L101-L128).

**Comment by sebastienvas** [comment]
Added aws image but did not delete the dockerfile as I don't know how it is used by Amazon team. Also updated the prow jobs by basically copying the one from prow. @fejta could you please check I did it right ?

File: boskos/BUILD.bazel [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package(default_visibility = ["//visibility:public"])
 
+load("@io_bazel_rules_docker//go:image.bzl", "go_image")
```
Comment (fejta) [inline boskos/BUILD.bazel]: Can we use prow_image so this has a good publish date?

File: boskos/BUILD.bazel [hunk 2]
```diff
@@ -13,6 +16,27 @@ go_binary(
...
+    name = "image",
+    embed = [":go_default_library"],
+    goarch = "amd64",
+    goos = "linux",
+    pure = "on",
+    visibility = ["//visibility:public"],
+)
+
+prow_push(
+    name = "release-push",
```
Comment (fejta) [inline boskos/BUILD.bazel]: maybe just push?

File: boskos/push.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+# * ensure there are no pending changes
+# * optionally activate GOOGLE_APPLICATION_CREDENTIALS and configure-docker if set
+# * run //boskos:release-push to build and push boskos images
+# * update all the cluster/*.yaml files to use the new image tags
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# TODO(fejta): rewrite this in a better language REAL SOON
```
Comment (fejta) [inline boskos/push.sh]: Might be good to have a TODO to make this script reusable

File: boskos/BUILD.bazel [hunk 4]
```diff
@@ -13,6 +16,27 @@ go_binary(
     pure = "on",
 )
 
+go_image(
```
Comment (fejta) [inline boskos/BUILD.bazel]: This will use a distroless image by default, is that okay?

Comment (sebastienvas) [reply boskos/BUILD.bazel]: yes that s fine.

File: boskos/push.sh (hunk 3 above)
Comment (sebastienvas) [reply boskos/push.sh]: done.

File: boskos/BUILD.bazel (hunk 2 above)
Comment (sebastienvas) [reply boskos/BUILD.bazel]: done.

File: boskos/BUILD.bazel (hunk 1 above)
Comment (sebastienvas) [reply boskos/BUILD.bazel]: done.

**Comment by sebastienvas** [comment]
@fejta PTAL

File: maintenance/aws-janitor/BUILD.bazel [hunk 5]
```diff
@@ -23,6 +24,16 @@ go_binary(
     visibility = ["//visibility:public"],
 )
 
+go_image(
```
Comment (fejta) [inline maintenance/aws-janitor/BUILD.bazel]: prow_image for this one too

**Pull request merged**